### PR TITLE
Add support for querying run events

### PR DESCRIPTION
**Pull request by glennsarti**
## Description

This commit adds API support for querying the List and Read of run-events.

## Testing plan

Integration tests are sufficient

## External links

- [API documentation](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#get-run-details)
- [Jira](https://hashicorp.atlassian.net/browse/TF-5613)

The API docs only tangentially refer to run-events as per the example response payload. Unfortunately the run-events API endpoint is currently undocumented.

## Output from tests

```
=== RUN   TestRunEventsList
=== RUN   TestRunEventsList/without_list_options
=== RUN   TestRunEventsList/with_all_includes
=== RUN   TestRunEventsList/without_a_valid_run_ID
--- PASS: TestRunEventsList (5.55s)
    --- PASS: TestRunEventsList/without_list_options (0.39s)
    --- PASS: TestRunEventsList/with_all_includes (0.47s)
    --- PASS: TestRunEventsList/without_a_valid_run_ID (0.00s)
=== RUN   TestRunEventsRead
=== RUN   TestRunEventsRead/without_read_options
=== RUN   TestRunEventsRead/with_all_includes
=== RUN   TestRunEventsRead/without_a_valid_run_event_ID
--- PASS: TestRunEventsRead (5.97s)
    --- PASS: TestRunEventsRead/without_read_options (0.32s)
    --- PASS: TestRunEventsRead/with_all_includes (0.31s)
    --- PASS: TestRunEventsRead/without_a_valid_run_event_ID (0.00s)
```

### CODE REVIEW

**Comment by glennsarti** [comment]
CI failures seem unrelated

File: run_event.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+	return rl, nil
+}
+
+// Read a run by its ID.
+func (s *runEvents) Read(ctx context.Context, runEventID string) (*RunEvent, error) {
+	return s.ReadWithOptions(ctx, runEventID, nil)
+}
+
+// Read a run by its ID with the given options.
```
Comment (brandonc) [inline run_event.go:105]:
```suggestion
// ReadWithOptions reads a run by its ID with the given options.
```

File: run_event.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+// RunEventList represents a list of run events.
+type RunEventList struct {
+	// Pagination is not supported by the API
+	*Pagination
+	Items []*RunEvent
+}
+
+// Run represents a Terraform Enterprise run.
```
Comment (brandonc) [inline run_event.go:43]:
```suggestion
// RunEvent represents a Terraform Enterprise run event.
```

File: run_event.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+// RunEventIncludeOpt represents the available options for include query params.
+type RunEventIncludeOpt string
+
+const (
+	RunEventComment RunEventIncludeOpt = "comment"
+	RunEventActor   RunEventIncludeOpt = "actor"
+)
+
+// RunListOptions represents the options for listing runs.
```
Comment (brandonc) [inline run_event.go:63]:
```suggestion
// RunEventListOptions represents the options for listing run events.
```

File: run_event.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+type RunEventListOptions struct {
+	// Optional: A list of relations to include. See available resources:
+	Include []RunEventIncludeOpt `url:"include,omitempty"`
+}
+
+// RunReadOptions represents the options for reading a run.
```
Comment (brandonc) [inline run_event.go:69]:
```suggestion
// RunEventReadOptions represents the options for reading a run event.
```

File: run_event.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+type RunEventReadOptions struct {
+	// Optional: A list of relations to include. See available resources:
+	Include []RunEventIncludeOpt `url:"include,omitempty"`
+}
+
+// List all the runs of the given workspace.
```
Comment (brandonc) [inline run_event.go:75]:
```suggestion
// List all the run events of the given run.
```

**Review by brandonc — CHANGES_REQUESTED** [review]
Can you also add a CHANGELOG entry?

File: run_event.go
Comment (brandonc) [inline run_event.go:1]: giphy

**Comment by sebasslash** [comment]
Please rebase with `main` to fix CI failures :+1:

**Review by brandonc — APPROVED** [review]

**Comment by glennsarti** [comment]
@brandonc Al good.  Thankyou!

**Pull request merged**